kornia/metrics/ssim.py

```diff
@@ -68,14 +83,29 @@ def ssim(
     mu1: torch.Tensor = filter2d(img1, kernel)
     mu2: torch.Tensor = filter2d(img2, kernel)
```

Comment:
```suggestion
    mu1: torch.Tensor = filter2d(img1, kernel, pad="valid")
    mu2: torch.Tensor = filter2d(img2, kernel, pad="valid")
```

I think applying here seem to be more reasonable to align with other libs.

Reply:
Yes, it is more reasonable... Oh, I see. Now in filter2d, there is a padding option now, before there is not. I will change it!

Another thing I also think is worth changing is that the gaussian 2d filter can be reduced to two 1d convolutions (I can see there already is a `filter2d_separable` function). It can reduce complexity if we use that instead of this `filter2d` function. Do you recommend me change it in the SSIM function?

Reply:
If the numerical precision is kept, why not

Reply:
> Yes, it is more reasonable... Oh, I see. Now in filter2d, there is a padding option now, before there is not. I will change it!
> 
> Another thing I also think is worth changing is that the gaussian 2d filter can be reduced to two 1d convolutions (I can see there already is a `filter2d_separable` function). It can reduce complexity if we use that instead of this `filter2d` function. Do you recommend me change it in the SSIM function?

Agree with edgar that if it accelerates without numerical changes. We can always upgrade.